src/objects.c

```diff
@@ -1340,28 +1334,64 @@ int p11prov_obj_export_public_ec_key(P11PROV_OBJ *obj, OSSL_CALLBACK *cb_fn,
     group = d2i_ECPKParameters(NULL, (const unsigned char **)&val,
```

Comment:
given we do this already in fetch_ec_key(), I wonder if we shouldn't simply get all this data at that time and store it in the object, d2i functions are not exactly light (many allocations and transformations) and the results never change, might be better to do the op once at fetch and not every time a param is requested.

Reply:
Yeah. That would be even better. I think storing the variable in `p11prov_key` should do that, but I dont have the whole architecture in my mind now so I am ok if you want to do it.

Not sure if we have this issue with other parameters as libssh uses only this one in this way now.